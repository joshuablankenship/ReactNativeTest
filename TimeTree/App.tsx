import 'react-native-gesture-handler';
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
//@ts-ignore
import {createStackNavigator} from '@react-navigation/stack';
import Login from './containers/Login';
import Main from './containers/Main';

export type RootStackParamList = {
  Login: undefined;
  Main: undefined;
};
const Stack = createStackNavigator<RootStackParamList>();
const App = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="Login"
          component={Login}
          options={{headerShown: false}}
        />
        <Stack.Screen
          name="Main"
          component={Main}
          options={{headerShown: false}}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
