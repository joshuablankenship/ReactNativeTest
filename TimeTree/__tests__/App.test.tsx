import {NavigationContainer} from '@react-navigation/native';
import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import App from '../App';

describe('Test App', () => {
  let wrapper: ShallowWrapper;
  wrapper = shallow(<App />);
  it('It should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('It should contain one navigation container', () => {
    expect(wrapper.find(NavigationContainer).length).toEqual(1);
  });
});
