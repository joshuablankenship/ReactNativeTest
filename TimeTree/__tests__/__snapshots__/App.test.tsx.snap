// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App It should match initial snapshot 1`] = `
<ForwardRef(NavigationContainer)>
  <StackNavigator>
    <Screen
      component={[Function]}
      name="Login"
      options={
        Object {
          "headerShown": false,
        }
      }
    />
  </StackNavigator>
</ForwardRef(NavigationContainer)>
`;

exports[`Test Login Screen It should match initial snapshot 1`] = `
<ForwardRef(NavigationContainer)>
  <StackNavigator>
    <Screen
      component={[Function]}
      name="Login"
      options={
        Object {
          "headerShown": false,
        }
      }
    />
  </StackNavigator>
</ForwardRef(NavigationContainer)>
`;
